File: 0165-compare-version-numbers/0165-compare-version-numbers.cpp
class Solution {
public:
    int compareVersion(string version1, string version2) {
        vector<string> tokens1, tokens2;
        stringstream iss1(version1), iss2(version2);
        string token;

        while (getline(iss1, token, '.')) tokens1.push_back(token);
        while (getline(iss2, token, '.')) tokens2.push_back(token);

        int i1, i2;
        for (int i = 0; i < max(tokens1.size(), tokens2.size()); ++i) {
            i1 = i < tokens1.size() ? stoi(tokens1[i]) : 0;
            i2 = i < tokens2.size() ? stoi(tokens2[i]) : 0;

            if (i1 != i2) return i1 > i2 ? 1 : -1;
        }
        // The versions are equal
        return 0;
    }
};

File: 0165-compare-version-numbers/0165-compare-version-numbers.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:

        v1 = version1.split('.')
        v2 = version2.split('.')

        v1_array = []
        v2_array = []

        for num_str in v1:
            v1_array.append(int(num_str.strip()))

        for num_str in v2:
            v2_array.append(int(num_str.strip()))

        # print(v1_array)
        # print(v2_array)

        maxLen = max(len(v1_array), len(v2_array))

        if len(v1_array) < maxLen:
           v1_array = v1_array + [0] * (maxLen - len(v1_array))

        if len(v2_array) < maxLen:
           v2_array = v2_array + [0] * (maxLen - len(v2_array))

        for n1, n2 in zip(v1_array, v2_array):
            if n1 > n2:
                return 1
            elif n1 < n2:
                return -1
            else:
                continue

        return 0
        
